packages/ampd-sdk/src/grpc/client.rs

```diff
@@ -15,21 +22,31 @@ pub enum Error {
 
     #[error("failed to execute gRPC request")]
     GrpcRequest(#[from] tonic::Status),
+
+    #[error("failed to convert event")]
+    EventConversion,
+
+    #[error("missing event in response")]
+    InvalidResponse,
 }
 
-#[automock]
+#[automock(type Stream = tokio_stream::Iter<vec::IntoIter<Result<Event, Error>>>;)]
 #[async_trait]
 #[allow(dead_code)]
 pub trait Client {
-    // TODO: This trait's methods should return our own types rather than the generated protobuf ones
+    type Stream: Stream<Item = Result<Event, Error>>;
+
     async fn subscribe(
-        &self,
-        request: SubscribeRequest,
-    ) -> Result<Streaming<SubscribeResponse>, Error>;
+        &mut self,
+        filters: Vec<AbciEventTypeFilter>,
+        include_block_begin_end: bool,
+    ) -> Result<Self::Stream, Error>;
 }
 
 #[allow(dead_code)]
+#[derive(Clone)]
 pub struct GrpcClient {
+    // TODO: separate trait (with automock) for blockchain and crypto to make testing easier
```

Comment:
I would remove this comment, traits (and mocks) should arise organically from a need, and not planned out upfront just because we think we need them.